pkg/engine/lifecycletest/update_plan_test.go

```diff
@@ -1640,3 +1640,289 @@ func TestPlannedUpdateWithDependentDelete(t *testing.T) {
+	// errors.
+
+	loaders := []*deploytest.ProviderLoader{
+		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{
+				Name: "pkgA",
+				CreateF: func(urn resource.URN, news resource.PropertyMap, timeout float64,
+					preview bool,
+				) (resource.ID, resource.PropertyMap, resource.Status, error) {
+					return resource.ID(urn.Name()), news, resource.StatusOK, nil
+				},
+				UpdateF: func(urn resource.URN, id resource.ID, olds, news resource.PropertyMap, timeout float64,
+					ignoreChanges []string, preview bool,
+				) (resource.PropertyMap, resource.Status, error) {
+					return news, resource.StatusOK, nil
+				},
+				CheckF: func(urn resource.URN,
+					olds, news resource.PropertyMap, _ []byte,
+				) (resource.PropertyMap, []plugin.CheckFailure, error) {
+					return news, nil, nil
+				},
+			}, nil
+		}),
+	}
+
+	program := deploytest.NewLanguageRuntime(func(_ plugin.RunInfo, monitor *deploytest.ResourceMonitor) error {
+		_, _, _, err := monitor.RegisterResource("pkgA:m:typA", "resA", true, deploytest.ResourceOptions{
+			Inputs: resource.PropertyMap{
+				"foo": resource.NewStringProperty("bar"),
+			},
+		})
+		assert.NoError(t, err)
+
+		_, _, _, err = monitor.RegisterResource("pkgA:m:typA", "resB", true, deploytest.ResourceOptions{
+			Inputs: resource.PropertyMap{
+				"foo": resource.NewStringProperty("bar"),
+			},
+		})
+		assert.NoError(t, err)
+
+		return nil
+	})
+
+	host := deploytest.NewPluginHost(nil, nil, program, loaders...)
+
+	p := &TestPlan{
+		Options: UpdateOptions{
+			Host:         host,
+			GeneratePlan: true,
+			Experimental: true,
+		},
+	}
+
+	project := p.GetProject()
+
+	// Resource plans get mutated during use
+	getPlan := func() *deploy.Plan {
+		// Run an update that creates A
+		plan, res := TestOp(Update).Plan(project, p.GetTarget(t, nil), UpdateOptions{
+			Host:         host,
+			GeneratePlan: true,
+			Experimental: true,
+			UpdateTargets: deploy.NewUrnTargets([]string{
+				"urn:pulumi:test::test::pkgA:m:typA::resB",
+				"urn:pulumi:test::test::pulumi:providers:pkgA::default",
+			}),
+		}, p.BackendClient, nil)
+		assert.Nil(t, res)
+		assert.NotNil(t, plan)
+		return plan
+	}
+
+	{
+		_, res := TestOp(Update).Run(project, p.GetTarget(t, nil), UpdateOptions{
+			Plan:         getPlan(),
+			Host:         host,
+			Experimental: true,
+		}, false, p.BackendClient, nil)
+		assert.NotNil(t, res)
+	}
+
+	{
+		_, res := TestOp(Update).Run(project, p.GetTarget(t, nil), UpdateOptions{
+			Plan:         getPlan(),
+			Host:         host,
+			Experimental: true,
+			UpdateTargets: deploy.NewUrnTargets([]string{
+				"urn:pulumi:test::test::pkgA:m:typA::resB",
+				"urn:pulumi:test::test::pulumi:providers:pkgA::default",
+			}),
+		}, false, p.BackendClient, nil)
+		assert.Nil(t, res)
+	}
+}
+
+func TestTargetedDoesNotCreateProvider(t *testing.T) {
```

Comment:
This test captures the change in behavior (i.e. fails on master and succeeds on this branch)

Reply:
This test isn't using updatePlans and so shouldn't be in update_plan_test.go